### DISCUSSION TITLE
Squash and merge - lost committer

### DISCUSSION THREAD

**Discussion by slawekjaranowski** [community/community#32934] [Pull Requests]
When use `Squash and merge` options from GUI, person which do such actions is lost.

In commit we have `GitHub` as committer.

```
Author:     Author of PR <....@users.noreply.github.com>
Commit:    GitHub <noreply@github.com>
```

But with `Rebase and merge` committer is preserved

```
Author:     Author of PR <....@users.noreply.github.com>
Commit:    Committer Name <committer@email...>
```

**Comment by slawekjaranowski** [comment]
any comments?

**Reply by bmarwell** [reply]
Works for me. GitHub did the squash, so it is the committer.

**Reply by slawekjaranowski** [reply]
So maybe person who initiate action should be added to commit message, eg as co-author or something else.
Yes I know I can edit squash message myself 😄

**Reply by HannesWell** [reply]
This behavior is bothering me as well, especially since the information is available.

If it would great if one could at least choose to use the actor as committer, even if that means signatures are not possible (see https://github.com/orgs/community/discussions/32934#discussioncomment-5776296) via a configuration option.

> Works for me. GitHub did the squash, so it is the committer.

Github also did the rebase when one clicks `Rebase and merge` but still the person who clicked is recorded as committer. If that's the metric, then `Rebase and merge` would be broken.

**Reply by bmarwell** [reply]
You're right. Manually adding "Co-Authored-By: name <Mail>" is still the only solution up to this day. 😐

**Comment by jmarshall** [comment]
I agree that this is annoying: the squash may have been done via GitHub, but it is the committing person who is responsible for the decision to merge. So it is them who should be recorded in `Commit[ter]`, not GitHub.

GitHub staff have commented in the past (see https://github.com/isaacs/github/issues/1368#issuecomment-491078292) that they do this so that they can **sign** the resulting commit. (The bug discussed in that thread is still present and still breaks author information to this day.)

**Reply by jmjoy** [reply]
Not many people care whether the commit is verified or not. GitHub fails to mark the committer for this trivial matter and does not provide any configurable options.

**Reply by HannesWell** [reply]
I can only second that.
At least an option to choose between a signed committ with Github as committer and an unsigned commit with the real person as committer would be great.

**Comment by jmjoy** [comment]
I like `squash and merge`, because it can make the commit log neater and can reference PR, but I also hate that `squash and merge ` rudely loses the Committer information and changes it to GitHub itself (`GitHub <noreply@github.com>`), which I think is an insult to the committer.

**Reply by jmjoy** [reply]
https://github.com/orgs/community/discussions/144271

**Reply by i386x** [reply]
@jmjoy Imho this is because the final commit message is fabricated. There is more:
* it adds a PR number enclosed between parentheses at the end of a commit message title (not an issue actually, but for me it was a surprise)
* it also changes the email address of `Author` to your primary email address you have set in your GitHub profile
* and the more painful one: when you edit your commit message and `git push -f` your topic branch, the fabricated commit message going to the base branch discards the changes, using the very first version text of a commit message (see details I described here: https://github.com/trustification/trustify-tests/issues/41)
